application/apps/indexer/session/src/handlers/observing/mod.rs

```diff
@@ -94,10 +94,17 @@ async fn run_source_intern<S: ByteSource>(
         }
         ParserType::Dlt(settings) => {
             let fmt_options = Some(FormatOptions::from(settings.tz.as_ref()));
+            let someip_metadata = match &settings.fibex_file_paths {
+                Some(paths) => {
+                    FibexSomeipMetadata::from_fibex_files(paths.iter().map(PathBuf::from).collect())
+                }
+                None => None,
+            };
```

Comment:
`and_then()` can be used here.
The people who are working on rust features are really working on reducing nesting :grinning: 
 
```rust
            let someip_metadata = settings.fibex_file_paths.as_ref().and_then(|paths| {
                FibexSomeipMetadata::from_fibex_files(paths.iter().map(PathBuf::from).collect())
            });
```

Reply:
yes, but sometimes in same way as c-devs used to reduce their var names.. ;-)